src/components/Modal/Modal.jsx

```diff
@@ -0,0 +1,254 @@
+import React, { cloneElement, useCallback, useEffect, useMemo } from "react";
+import ReactDOM from "react-dom";
+import PropTypes from "prop-types";
+import cx from "classnames";
+import styles from "./Modal.module.scss";
+import { useA11yDialog } from "./a11yDialog";
+import { ModalContent, ModalFooter, ModalHeader } from "components";
+import { clearAllBodyScrollLocks, disableBodyScroll, enableBodyScroll } from "body-scroll-lock";
+import useAnimationProps from "components/Modal/useAnimationProps";
+import { useKeyEvent } from "../../hooks";
+
+export const MODAL_WIDTH = {
+  DEFAULT: "default",
+  FULL_WIDTH: "full_width"
+};
+
+const isModalHeader = child => child.type === ModalHeader;
+const isModalContent = child => child.type === ModalContent;
+const isModalFooter = child => child.type === ModalFooter;
+
+const Modal = ({
+  classNames,
+  id,
+  show,
+  title,
+  description,
+  onClose,
+  isAlertDialog,
```

Comment:
We are a bit strict about maintaining uniformity between the api of all the props of all the components. We do not used in isPrefix while naming props (because of React shorthand support for boolean props).
please change isAlertDialog to alertDialog.

Reply:
👍